### DISCUSSION TITLE
Listener trigger on same value

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#123] [General]
Is it normal/okay that a listener is executing even if the capsule value doesn't change?

I see this while I'm testing the "possible bug".

```dart
Future<void> main() async {
  final _ = cc.listen((use) {
    final wsState = use(websocketStateCapsule);
    print('Websocket state: ${wsState.asString}');
  });

  final __ = cc.listen((use) {
    final name = use(nameCapsule).value;
    print('Name: $name');
  });

  cc.read(websocketControllerCapsule).connect();
}
```

```dart
llib.CachedSerializableValueController<String> nameCapsule(
  CapsuleHandle use,
) =>
    use.cachedSerializableValue<String>(
      null,

      // Reseting on websocket disconnected
      doReset: use(websocketStateCapsule).disconnected,

      debugResetFlag: 'nameCapsule',
    );
```

Output:
![imagen](https://github.com/GregoryConrad/rearch-dart/assets/73592852/30a9067e-b8f2-4cec-a932-f6ff5f6ba432)

**Comment by busslina** [comment]
Okay, I'm wrong here. Capsule value is changing. Capsule value value is what is not changing.

> Is it normal/okay that a listener is executing even if the capsule value doesn't change?

I assume that the listener shouldn't execute if the capsule value doesn't change

**Reply by busslina** [reply]
This happened to me because I use Classes instead Records as my capsules controller. I use them because VSCode seems doesn't like Records and it is more difficult to track usage... But I just realize that using Record is better due to `==` operator will save me some rebuilds.

Am I correct?

**Reply by GregoryConrad** [reply]
Yup, that sounds correct
